src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -98,6 +80,53 @@ func getDirectoriesToDelete(keepSystemImages bool) ([]string, error) {
 	return dirs, nil
 }
 
+func GetLockfilePath(appName string) (string, error) {
+	appData, err := getKnownFolder(windows.FOLDERID_RoamingAppData)
+	if err != nil {
+		return "", fmt.Errorf("could not get AppData folder: %w", err)
+	}
+	return path.Join(appData, appName, "lockfile"), nil
+}
+
+func UnregisterWSL() error {
+	cmd := exec.Command("wsl", "--list", "--quiet")
+	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: 0x08000000} // CREATE_NO_WINDOW
+	rawBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	decoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewDecoder()
+	cookedBytes, err := decoder.Bytes(rawBytes)
```

Comment:
Doing `decoder.String(string(rawBytes))` might be easier?

Reply:
Actually I prefer specifying what I expect to see with the byte-order and BOM there.

Reply:
ok